core/drand_beacon_public.go

```diff
@@ -189,7 +189,7 @@ func (bp *BeaconProcess) SignalDKGParticipant(ctx context.Context, p *drand.Sign
 	bp.state.Lock()
```

Comment:
(VERY minor) wouldn't it be better to put this below the `if`, and remove the `Unlock` within the `if`?